### ISSUE TITLE
Code not compatible with the open-domain settings

### ISSUE THREAD

**Issue by xwhan** [shmsw25/qa-hard-em#1]
Thanks for sharing the code! The current version does not seem to be compatible with the open-domain experiments. I tried to fix some trivial errors, but was still unable to get the reported numbers. Are the dev splits same as the ones described in the original paper? 
Also, as the open-domain experiments are expensive to run, can you release the processed data for the non-open settings?

**Comment by shmsw25** [comment]
Hi @xwhan, thanks for your interest and sorry to hear you had troubles. What do you mean by it was not compatible and you had to fix some trivial errors? 
Yes the data splits are the same, which are released in README here as well. Could you give more details (e.g. which dataset are you trying, did you have trouble with MML objective or hard EM objective or both, which batch size and tau values did you use, etc.)

**Comment by xwhan** [comment]
I tried running the code and encourted the following errors:

1. https://github.com/shmsw25/qa-hard-em/blob/master/evaluate_qa.py#L188 
I suppose pred.paragraph_index should be an integer not list

2. https://github.com/shmsw25/qa-hard-em/blob/master/prepro.py#L53
the "rel_features" does not exists

I am tried both NaturalQuestions and TriviaQA with the MML objective and batch size 64 due to memory limit.

In the original paper, you mentioned that you used the dev set for testing and a portion of the training data for development. However, I only find a dev split in the preprocessed data. Is that the test set you used?

**Comment by xwhan** [comment]
I noticed that the test files are actually in the unprocessed data google drive folder. I also got reasonable results on TriviaQA-Open after fixing a preprocessing bug on my side due to parallel processing:
with batch size 64, MML objective and evaluation every 1000 updates, the best dev numbers are 49.59EM/54.82F1
I am closing the issue at this point, thanks again for the nice work~
